File: webtester/java/src/test/java/com/mytesting/pages/SearchResults.java
package com.mytesting.pages;

import java.util.List;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;

public class SearchResults extends BasePage {
 
    @FindBy(how=How.CSS, using="h4")
    private List<WebElement> allItems;

	@FindBy(how=How.XPATH, using="//h4/../..")
	private List<WebElement> allItemLinks;
	
    @FindBy(how=How.CSS, using="div[class^='ProductListContainer__noProducts']")
    private WebElement noItemsFoundMessage;
    
    @FindBy(how=How.CSS, using="div[class^='ProductListContainer__noContent']")
    private WebElement noItemsContainer;
	
    @FindBy(how=How.XPATH, using="//div[contains(@class,'Box product-page__user-price')]/h3/span")
    private WebElement productPrice; 
    
    @FindBy(how=How.CSS, using="span[class^='product-page__backButton']")
    private WebElement productPageHomeButton;
   
	public SearchResults(WebDriver driver) {
		super(driver);
	}
	
	public short getNumberOfItemsWithTitle(String title) {
		
		if(allItems.size() == 0)
			return 0;
		
		short numItemsWithTitle = 0;
		
		for( WebElement element: allItems) {
			
			actionBot.waitToBeDisplayed(element);
			
			if(element.getText().contains(title))
				numItemsWithTitle ++;		
		}
		
		return numItemsWithTitle;
	}
	
	public void clickFirstItemLink() {
		
		WebElement firstItemLink = allItemLinks.listIterator().next();
		advancedActionBot.waitToBeEnabledAndClick(firstItemLink);
	}
	
	public String getItemPrice() {
		
		actionBot.waitToBeEnabled(productPrice);
		return productPrice.getText();
		
	}
	
	public void waitForNoItemsFoundMessageVisible() {
		
		actionBot.waitToBeDisplayed(noItemsFoundMessage);
	}
	
	public void waitForEmptyResultsContainer() {
		
		actionBot.waitToBeDisplayed(noItemsContainer);
	}

}

File: webtester/nodejs/features/page_objects/search_results.js
const By = require('selenium-webdriver').By;
const until = require('selenium-webdriver').until;

var searchResults = function () {

    const elements = {
        itemTitleXpath: "//h4",
        itemTitleLocator: By.css("h4"),
        itemLinkLocator: By.xpath("//h4/../.."),
        noItemsFoundMessage: By.css('div[class^="ProductListContainer__noProducts"]'),
        noItemsContainer: By.css('div[class^="ProductListContainer__noContent"]'),
        productPriceLocator: By.xpath('//div[contains(@class,"Box product-page__user-price")]/h3/span'),
        productPageHomeButton: By.css('span[class^="product-page__backButton"]')
    };

    searchResults.prototype.getRelevantItemTitle = function (context, titleWord) {

        const itemLocator = By.xpath(elements.itemTitleXpath + "[contains(text(),'" + titleWord + "')]");
        context.driver.wait(until.elementLocated(itemLocator));
        return context.driver.findElement(itemLocator).getText();
    };

    searchResults.prototype.getFirstResultItemTitle = function (context) {
        context.driver.wait(until.elementLocated(elements.itemTitleLocator));
        return context.driver.findElement(elements.itemTitleLocator).getText();
    };

    searchResults.prototype.waitForNoItemsFoundMessageVisible = function (context) {

        context.driver.wait(until.elementLocated(elements.noItemsFoundMessage));
        const element = context.driver.findElement(elements.noItemsFoundMessage);
        return context.driver.wait(until.elementIsVisible(element));
    };

    searchResults.prototype.waitForEmptyResult = function (context) {

        return context.driver.wait(until.elementLocated(elements.noItemsContainer));
    };

    searchResults.prototype.clickFirstItemLink = function (context) {
        context.driver.wait(until.elementLocated(elements.itemLinkLocator));
        const element = context.driver.findElement(elements.itemLinkLocator);
        context.driver.wait(until.elementIsVisible(element));
        return element.click();
    };

    searchResults.prototype.getProductPrice = function (context) {
        context.driver.wait(until.elementLocated(elements.productPriceLocator));
        return context.driver.findElement(elements.productPriceLocator).getText();
    };

    searchResults.prototype.clickProductPageHomeButton = function (context) {
        context.driver.wait(until.elementLocated(elements.clickProductPageHomeButton));
        return context.driver.findElement(elements.clickProductPageHomeButton).click();
    };


};

module.exports = new searchResults();